libs/ddd-angular/src/helpers/domain-helper.ts

```diff
@@ -1,19 +1,97 @@
 import { Tree, readProjectConfiguration } from '@nrwl/devkit';
+import { DDDObject } from '../model/ddd.model';
+import { Domain } from '../model/domain.model';
+import { API } from '../model/api.model';
+import { Shell } from '../model/shell.model';
+import { Feature } from '../model/feature.model';
+import { UI } from '../model/ui.model';
+import { Util } from '../model/util.model';
 
+/**
+ * It throws an error if the domain is invalid or shared
+ *
+ * @param tree The virtual filesystem tree provided by NX
+ * @param domain The domain name
+ */
 export const guardValidDomain = (tree: Tree, domain: string) => {
   if (!domainExist(tree, domain)) throw new Error('Invalid domain');
   if (domain === 'shared') throw new Error('Shared cannot be a domain');
 };
 
+/**
+ * It checks if the project is a domain by checking if it has the tag 'type:domain-logic'
+ * It throws an error if the domain is invalid
+ *
+ * @param tree The virtual filesystem tree provided by NX
+ * @param domain The domain name
+ * @returns if the project is a domain
+ */
 export const domainExist = (tree: Tree, domain: string) => {
   const tags = readProjectConfiguration(tree, domain).tags;
   return tags?.includes('type:domain-logic');
 };
 
-export const domainNameFromProject = (tree: Tree, projectName: string) => {
-  const domainName = readProjectConfiguration(tree, projectName)
-    .tags?.find((tag) => tag.startsWith('domain:'))
-    ?.split(':')[1];
-  if (!domainName) throw new Error('Invalid domain');
-  return domainName;
+/**
+ * It finds the domain name from the tag 'domain:{domain-name}'
+ * It throws an error if the domain is invalid
+ *
+ * @param tree The virtual filesystem tree provided by NX
+ * @param projectName The project name
+ * @returns The domain name
+ */
+export const domainNameFromProject = (tree: Tree, projectName: string) =>
+  domainTagFromProject(tree, projectName).split(':')[1];
+
+/**
+ * It finds the domain tag from the tag starting with 'domain:'
+ * It throws an error if the domain is invalid
+ *
+ * @param tree The virtual filesystem tree provided by NX
+ * @param projectName The project name
+ * @returns the domain tag containing the domain name
+ */
+export const domainTagFromProject = (tree: Tree, projectName: string) => {
+  const domainTag = readProjectConfiguration(tree, projectName).tags?.find(
+    (tag) => tag.startsWith('domain:')
+  );
+  if (!domainTag) throw new Error('Invalid domain');
```

Comment:
The message should say why it threw an error.
ie. 'Invalid domain. Found tags "library:asdf", expected to start with "domain:"'